src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -440,6 +394,67 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeLong(ttl);
             out.writeByte(versionType.getValue());
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            Create create = (Create) o;
+
+            if (timestamp != create.timestamp) {
+                return false;
```

Comment:
This might be more readable and succinct with `&&` chaining, what do you think?

```java
return timestamp == create.timestamp &&
    ttl == create.ttl &&
    version == create.version &&
    id.equals(create.id) &&
    type.equals(create.type) &&
    ... etc ...
```